internal/provider/resource_packer_channel.go

```diff
@@ -0,0 +1,381 @@
+			"author_id": {
+				Description: "The author of the channel.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"iteration": {
+				Description: "The iteration assigned to the channel.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fingerprint": {
+							Description: "The fingerprint of the iteration assigned to the channel.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"id": {
+							Description: "The id of the iteration assigned to the channel.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"incremental_version": {
+							Description: "The incremental_version of the iteration assigned to the channel.",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"organization_id": {
+				Description: "The ID of the organization this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "Creation time of this build.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourcePackerChannelRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := d.Get("bucket_name").(string)
+	client := meta.(*clients.Client)
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+	if err := setLocationData(d, loc); err != nil {
+		return diag.FromErr(err)
+
+	}
+	// TODO (nywilken) may need to look into pagination depending on channel list sizes
+	resp, err := clients.ListBucketChannels(ctx, client, loc, bucketName)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	channelName := d.Get("name").(string)
+	var channel packermodels.HashicorpCloudPackerChannel
+	for _, c := range resp.Channels {
+		if c.Slug == channelName {
+			channel = *c
+			break
+		}
+	}
+	if channel.ID == "" {
+		return diag.Errorf("Unable to find channel in bucket %s named %s.", bucketName, channelName)
+	}
+	return setPackerChannelResourceData(d, &channel)
+}
+
+func resourcePackerChannelCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := d.Get("bucket_name").(string)
+	channelName := d.Get("name").(string)
+
+	client := meta.(*clients.Client)
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+	if err := setLocationData(d, loc); err != nil {
+		return diag.FromErr(err)
+	}
+
+	channel, err := clients.CreateBucketChannel(ctx, client, loc, bucketName, channelName)
```

Comment:
Any specific reason why you always create it empty? You could provide the iteration assignment at creation as well and assign it at the same time.

Reply:
I had it like this before and then changed to consolidate the assignment code. But now that it is all wired up maybe I can do the assignment within the create, as well.

Reply:
done